addons/api/addon/serializers/auth-method.js

```diff
@@ -17,19 +17,35 @@ export default class AuthMethodSerializer extends ApplicationSerializer {
    */
   serialize(snapshot) {
     switch (snapshot.record.type) {
-      case TYPE_AUTH_METHOD_OIDC:
-        return this.serializeOIDC(...arguments);
+      case TYPE_AUTH_METHOD_PASSWORD:
+        return this.serializePassword(...arguments);
       default:
         return this.serializeDefault(...arguments);
     }
   }
 
+  serializeAttribute(snapshot, json, key, attribute) {
+    super.serializeAttribute(...arguments);
+    const { type } = snapshot.record;
+    const { options } = attribute;
+
+    // If an attribute has a `for` option, it must match the
+    // record's `type`, else the attribute excluded
+    // from serialization.
+    if (options?.for && options.for !== type && !options.for.includes(type)) {
```

Comment:
since type is a string, 
```suggestion
   if (options?.for && !options.for.includes(type)) {
```